crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -646,6 +646,11 @@ impl<F: Clone> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for Paymen
             })
             .await;
 
+        let merchant_order_reference_id = payment_data
```

Comment:
this update is redundant, since we are inserting it while creating. And reference id does not change in the flow